File: apps/backend/app/schemas/app_config.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field

ConfigValue = int | float
ConfigGroup = Literal["analytics", "workspace"]


class AppConfigItem(BaseModel):
    key: str
    value: ConfigValue
    default_value: ConfigValue
    description: str
    group: ConfigGroup


class AppConfigResponse(BaseModel):
    items: list[AppConfigItem]


class AppConfigUpdateItem(BaseModel):
    key: str
    value: ConfigValue


class AppConfigUpdateRequest(BaseModel):
    items: list[AppConfigUpdateItem] = Field(min_length=1)

File: apps/frontend/src/api/modules/app-config.ts
import { apiRequest } from '@/api/http-client';

export type AppConfigGroup = 'analytics' | 'workspace';

export interface AppConfigItem {
  key: string;
  value: number;
  default_value: number;
  description: string;
  group: AppConfigGroup;
}

export interface AppConfigResponse {
  items: AppConfigItem[];
}

export interface AppConfigUpdateRequest {
  items: Array<{
    key: string;
    value: number;
  }>;
}

export function getAppConfig(): Promise<AppConfigResponse> {
  return apiRequest<AppConfigResponse>('/api/v1/config');
}

export function updateAppConfig(payload: AppConfigUpdateRequest): Promise<AppConfigResponse> {
  return apiRequest<AppConfigResponse>('/api/v1/config', {
    method: 'PATCH',
    body: JSON.stringify(payload),
  });
}

export function resetAppConfig(): Promise<AppConfigResponse> {
  return apiRequest<AppConfigResponse>('/api/v1/config/reset', {
    method: 'POST',
  });
}
